app/src/main/java/ar/com/wolox/android/example/ui/home/news/NewsHolder.kt

```diff
@@ -4,26 +4,29 @@ import android.support.v7.widget.RecyclerView
 import android.view.View
 import ar.com.wolox.android.R
 import ar.com.wolox.android.example.model.New
+import ar.com.wolox.android.example.utils.DateUtils
 import kotlinx.android.synthetic.main.item_new.view.*
-import org.joda.time.format.DateTimeFormat
-import org.ocpsoft.prettytime.PrettyTime
-import java.util.Locale
 
-class NewsHolder(view: View) : RecyclerView.ViewHolder(view) {
+class NewsHolder(view: View, private val mOnNewClickListener: INewClickListener) : RecyclerView.ViewHolder(view) {
+
+    private lateinit var mNew: New
+    var mUserLoggedIn: Int = -1
+
+    init {
+        itemView.setOnClickListener {
+            mOnNewClickListener.onNewClicked(mNew)
+        }
+    }
 
     fun bindHolder(item: New) {
-        itemView.mNewTitle.text = item.title
-        itemView.mNewText.text = item.text
+        mNew = item
 
-        val localDate = DateTimeFormat
-                .forPattern(itemView.context.getString(R.string.app_date_format))
-                .withLocale(Locale.getDefault())
-                .parseLocalDate(item.createdAt)
-                .toDate()
+        itemView.mNewTitleText.text = item.title
+        itemView.mNewText.text = item.text
 
-        itemView.mTimeAgo.text = PrettyTime().format(localDate)
+        itemView.mTimeAgoText.text = DateUtils.toDuration(item.createdAt, itemView.context)
 
-        itemView.mLikeIcon.setImageResource(R.drawable.ic_like_off)
+        itemView.mLikeIcon.setImageResource(if (item.likes.contains(mUserLoggedIn)) R.drawable.ic_like_on else R.drawable.ic_like_off)
```

Comment:
Where is set mUserLoggedIn? It's my opinion but I don't like the idea to have a default mUserLoggedIn = -1. I mean, the user id mustn't be null.

Reply:
The member field `mUserLoggedIn` is initialized with a non-null value by the `NewsAdapter` instance

Reply:
So if it is non-null, why you're assigning a -1 value?

Reply:
Fixed, passing userId on constructor of holder